### PR TITLE
chore(main): Added artifacts definition in release job

### PR DESCRIPTION
**Pull request by diptanshumittal**
Updating the Kokoro release job for storing the artifacts released on rubygems.org in placer.

### CODE REVIEW

File: .kokoro/release.cfg [hunk 1]
```diff
@@ -31,3 +31,12 @@ env_vars: {
   key: "SECRET_MANAGER_KEYS"
   value: "releasetool-publish-reporter-app,releasetool-publish-reporter-googleapis-installation,releasetool-publish-reporter-pem,docuploader_service_account"
 }
+
+# Store the packages uploaded to rubygems.org, which
+# we can later use to generate SBOMs and attestations.
+action {
+ define_artifacts {
+   regex: "github/ruby-spanner/**/pkg/*.gem"
```
Comment (dazuma) [inline .kokoro/release.cfg:39]: I think you can use `github/ruby-spanner/*/pkg/*.gem`. There should always be only one directory level for a gem in this repo. Might allow us to avoid an edge case where a gem archive shows up in, for example, a test directory.

**Review by dazuma — CHANGES_REQUESTED** [review]

File: .kokoro/release.cfg (hunk 1 above)
Comment (dazuma) [reply .kokoro/release.cfg]: For newer repos related to client libraries, we tend to put the gem inside a subdirectory anyway even if it's the only one. This is because:

* It makes naming less awkward because the gem name is seldom the same name as the repo. (Ruby repos should have the word "ruby" in the repo name, but the word "ruby" should not show up in the gem name.)
* It hedges against the possibility that we want to add additional closely related gems in the future. 
* It doesn't hurt us, since our tooling is designed to handle monorepos
